src/ovs_events.c

```diff
@@ -0,0 +1,629 @@
+ */
+static ovs_events_ctx_t ovs_events_ctx = {
+    .mutex = PTHREAD_MUTEX_INITIALIZER,
+    .config = {.send_notification = 0,     /* do not send notification */
+               .ovs_db_node = "localhost", /* use default OVS DB node */
+               .ovs_db_serv = "6640",      /* use default OVS DB service */
+               .ifaces = NULL},
+    .ovs_db_select_params = NULL,
+    .is_db_available = 0,
+    .ovs_db = NULL};
+
+/* This function is used only by "OVS_EVENTS_CTX_LOCK" define (see above).
+ * It always returns 1 when context is locked.
+ */
+static int ovs_events_ctx_lock() {
+  pthread_mutex_lock(&ovs_events_ctx.mutex);
+  return (1);
+}
+
+/* This function is used only by "OVS_EVENTS_CTX_LOCK" define (see above).
+ * It always returns 0 when context is unlocked.
+ */
+static int ovs_events_ctx_unlock() {
+  pthread_mutex_unlock(&ovs_events_ctx.mutex);
+  return (0);
+}
+
+/* Check if given interface name exists in configuration file. It
+ * returns 1 if exists otherwise 0. If no interfaces are configured,
+ * -1 is returned
+ */
+static int ovs_events_config_iface_exists(const char *ifname) {
+  if (ovs_events_ctx.config.ifaces == NULL)
+    return -1;
+
+  /* check if given interface exists */
+  for (ovs_events_iface_list_t *iface = ovs_events_ctx.config.ifaces; iface;
+       iface = iface->next)
+    return (strcmp(ifname, iface->name) == 0);
+
+  return 0;
+}
+
+/* Get OVS DB select parameter request based on rfc7047,
+ * "Transact" & "Select" section
+ */
+static char *ovs_events_get_select_params() {
+  int ret = 0;
+  size_t buff_size = 0;
+  size_t offset = 0;
+  char *buff = NULL;
+  char *new_buff = NULL;
+  const char params_fmt[] = "[\"Open_vSwitch\"%s]";
+  const char option_fmt[] = ",{\"op\":\"select\",\"table\":\"Interface\","
+                            "\"where\":[[\"name\",\"==\",\"%s\"]],"
+                            "\"columns\":[\"link_state\",\"external_ids\","
+                            "\"name\",\"_uuid\"]}";
+  const char default_opt[] = ",{\"op\":\"select\",\"table\":\"Interface\","
+                             "\"where\":[],\"columns\":[\"link_state\","
+                             "\"external_ids\",\"name\",\"_uuid\"]}";
+  /* setup OVS DB interface condition */
+  for (ovs_events_iface_list_t *iface = ovs_events_ctx.config.ifaces; iface;
+       iface = iface->next, offset += ret) {
+    /* allocate new buffer (format size + ifname len is good enough) */
+    buff_size += (sizeof(option_fmt) + strlen(iface->name));
+    new_buff = realloc(buff, buff_size);
+    if (new_buff == NULL)
+      goto failure;
+    buff = new_buff;
+    ret = ssnprintf(buff + offset, buff_size, option_fmt, iface->name);
+    if (ret < 0)
+      goto failure;
+  }
+  /* if no interfaces are configured, use default params */
+  if (buff == NULL) {
+    buff = strdup(default_opt);
+    offset = strlen(default_opt);
+  }
+
+  /* allocate memory for OVS DB select params */
+  buff_size = offset + sizeof(params_fmt);
```

Comment:
I'm not a huge fan of re-using variables like this. Please consider using `char *params` and `size_t params_size` instead.